glue/flagging_script_glue/flagging_e894e5.py

```diff
@@ -1085,20 +1095,14 @@ def name_selector(tokens) -> str:
         'Empty Name' if name is empty.
         id (str): identified last name
     """
-    if tokens == "Empty Name":
-        return tokens
-    # Ex: John Smith Jr
+
+    if tokens == "Empty Name" or tokens == []:
+        return "Empty Name"
+
     if tokens[-1] in ["jr", "sr", "ii", "iii", "iv", "v"]:
         tokens = tokens[:-1]
-    # Ex: John Smith
-    if len(tokens) == 2:
-        id = tokens[1]
-    # John George Smith
-    if len(tokens) == 3:
-        id = tokens[2]
-    # John George Theodore Smith
-    else:
-        id = tokens[-1]
+
+    id = tokens[-1]
```

Comment:
We simplify the logic here as the prior code in this function was redundant (grabbing last element of list)